### DISCUSSION TITLE
Embedding javascript snippet into html when theia loads

### DISCUSSION THREAD

**Discussion by pgirishibm** [eclipse-theia/theia#8726] [Q&A]
I need to integrate walkme (https://www.walkme.com/)  into the theia UI.  To integrate Walkme requires to add a Javascript snippet into the html page that is being rendered. In case of  theia, we do not have control over the html page and embedding code is not simple.

Is there any way code snippet can be inserted when the preprocessor generates HTML pages for  Theia ?

I see a similar question asked for che-theia. https://github.com/eclipse/che/issues/18073

**Comment by kittaakos** [comment]
> In case of theia, we do not have control over the html page and embedding code is not simple.

In the end, it's just JS and some HTML. Replace the `index.html`, or any other generated JS modules after the `theia build` step. I do this almost on every project that requires customization.

**Reply by pgirishibm** [reply]
@kittaakos which is the index.html that is part of the che-theia build that i can use to embed the JS snippet? In Eclipse che, che-theia is embedded into an iframe which is part of the che-dashboard. but che-thia by itself does not have its own index.html right?   

For Walkme, the code needs to be embedded into the html file that is rendered in the browser.

**Reply by kittaakos** [reply]
> which is the index.html that is part of the che-theia
Please ask on a Che forum.
